model/riscv_insts_zicboz.sail

```diff
@@ -0,0 +1,66 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+// Cache Block Operations - Zero
+//
+// Note that the CBOP prefetch instructions are existing base instructions that
+// are just defined to be hints, so there is not need to add any code for them
+// (since this model has no caches to prefetch into and the encodings already
+// exist).
+
+function cbo_zero_enabled(p : Privilege) -> bool = feature_enabled_for_priv(p, menvcfg.CBZE()[0], senvcfg.CBZE()[0])
+
+/* ****************************************************************** */
+union clause ast = RISCV_ZICBOZ : (regidx)
+
+mapping clause encdec = RISCV_ZICBOZ(rs1)                if haveZicboz()
+  <-> 0b000000000100 @ rs1 @ 0b010 @ 0b00000 @ 0b0001111 if haveZicboz()
+
+mapping clause assembly = RISCV_ZICBOZ(rs1)
+  <-> "cbo.zero" ^ spc() ^ "(" ^ opt_spc() ^ reg_name(rs1) ^ opt_spc() ^ ")"
+
+function clause execute(RISCV_ZICBOZ(rs1)) = {
+  if cbo_zero_enabled(cur_privilege) then {
+    let rs1_val = X(rs1);
+    let cache_block_size_exp = plat_cache_block_size_exp();
+    let cache_block_size = 2 ^ cache_block_size_exp;
+
+    // Offset to the beginning of the cache block.
+    let offset = (rs1_val & ~(zero_extend(ones(cache_block_size_exp)))) - rs1_val;
+
+    match ext_data_get_addr(rs1, offset, Write(Data), cache_block_size) {
```

Comment:
```suggestion
    // Offset from the beginning of the cache block.
    let offset = rs1_val - (rs1_val & ~(zero_extend(ones(cache_block_size_exp))));

    match ext_data_get_addr(rs1, -offset, Write(Data), cache_block_size) {
```
Not sure if this is any better, but I find it easier to understand if we subtract the offset in the call to ext_data_get_addr.

Reply:
Actually... there isn't a unary `-` operator for bit vectors so you have to do `zeros() - offset` which is just a bit weird, so I changed it back.